### ISSUE TITLE
Firebase v9 auth and firestore does not work and endless promise rejection errors occur.

### ISSUE THREAD

**Issue by jordanlee-dev** [firebase/firebase-js-sdk#5444]
### [REQUIRED] Describe your environment

  * Operating System version:  webOS TV 2.x
  * Browser version: Chromium, web engine: WebKit 538.2
  * Firebase SDK version: 9.0.1 (modular)
  * Firebase Product: auth, firestore
  * App framework: React-Js
  * Node version: 14.16.1

#### Steps to reproduce:

**With v8 these problems did not exist.** As you can see in the image below with V9, it gives an error on the line "undefined is not a function (evaluating 'fn(e, !1).getAll()')" and it starts giving an endless promise rejection error. It's not auth and I can't get data from firestore. In short, none of them work.
These errors in my project do not exist in other browsers on windows 10. It's only on webos.

webos browser user-agent: Mozilla/5.0 (Web0S; Linux/SmartTV) AppleWebKit/538.2 (KHTML, like Gecko) Large Screen WebAppManager Safari/538.2

**In my same application there are no problems and errors on Windows10.**

![firebase error](https://user-images.githubusercontent.com/80259148/132089443-d6c4da84-90af-4be9-9ab0-1ad87c296d0d.png)

#### Relevant Code:

```javascript

// Error happened firebase code 
...............
key: "_poll",
              value: (function () {
                var e = Object(d.a)(
                  f.a.mark(function e() {
                    var t, n, r, i, a, o, s, u, l, h, d;
                    return f.a.wrap(
                      function (e) {
                        for (;;)
                          switch ((e.prev = e.next)) {
                            case 0:
                              return (
                                (e.next = 2),
                                this._withRetries(function (e) {
                                  var t = fn(e, !1).getAll(); //*****There is an error here
                                  return new ln(t).toPromise();
                                })
                              );
...........
```

**Comment by hsubox76** [comment]
Can you share the relevant lines from your code? Which Auth and Firestore methods are you calling?

**Comment by jordanlee-dev** [comment]
I just made a very simple example with auth. Unfortunately same problem again. As you can see in the picture, it gives an infinite rejection error. It receives the auth information, that is, it provides the connection, but there are errors. As you can see I didn't use Firestore.
No errors in Windows10. Only webos gives an error.

The relevant codes are below

![firebasesimple_auth](https://user-images.githubusercontent.com/80259148/132403263-549105f7-9419-4564-a4be-fad90111a3d3.png)

```javascript

// firebase.js
import { initializeApp } from "firebase/app";

import { firebaseConfig } from "./config";

import {
  getAuth,
  onAuthStateChanged,
  signInWithEmailAndPassword,
} from "firebase/auth";

const firebaseApp = initializeApp(firebaseConfig);

const auth = getAuth(firebaseApp);

export { auth, onAuthStateChanged, signInWithEmailAndPassword };
```


```javascript

// app.js

import React, { useEffect } from "react";

import MainMenu from "./components/mainMenu";

import {
  auth,
  signInWithEmailAndPassword,
} from "./components/firebase/firebase";

import "./App.css";

function App() {
  const signInTest = async (email, password) => {
    try {
      const response = await signInWithEmailAndPassword(
        auth,
        "userEmail",
        "userPassword"
      );

      console.log("user Info: ", response.user);

      return response.user;
    } catch (error) {
      console.log("signIN error: ", error);
    }
  };
  useEffect(() => {
    console.log("start");

    signInTest();

    return () => {
      console.log("unmount");
    };
  }, []);

  return <MainMenu />;
}

export default App;

```

**Comment by jordanlee-dev** [comment]
When connecting as follows, with initializeAuth and persistence it gives a promise rejection error only once, but this time it says internet problem...

and

**Failed to get document because the client is offline**

I think it's a browser issue

```javascript

...............

// const firebaseAuth = getAuth(firebaseApp);

const firebaseAuth = initializeAuth(firebaseApp, {
  persistence: browserLocalPersistence,
});
...............

```


![FirebaseErrorNew](https://user-images.githubusercontent.com/80259148/132904900-56aad98f-8fd7-44fd-8bcf-00520485b9ec.png)

**Comment by sureshjoshi** [comment]
Is there any update on this one? I'm running into the same issue on Microsoft Edge Legacy on Windows 10. Basically had to tear my app apart to find what was causing the never ending "Unhandled promise rejection TypeError: Object doesn't support property or method 'getAll'" sitting inside the same `withRetries` as above.

It all came down to this call: `const auth = getAuth(firebaseApp);`

I thought I have all the polyfills and es5 transpilation in place (everything else works), but no luck yet.

**Comment by sureshjoshi** [comment]
So, it turns out the problem is that IndexedDB getAll is not supported on IE (https://developer.mozilla.org/en-US/docs/Web/API/IDBIndex/getAll)

Knowing that, it appears to be pretty simple to resolve by installing this shim: https://github.com/dumbmatter/IndexedDB-getAll-shim 

I'm not sure if this is documented somewhere, as I saw the required polyfills in the docs, but I don't recall seeing anything about the IndexedDB polyfill. Alternatively, if Firebase came with a shim, that would be handy too.

**Comment by hsubox76** [comment]
Thanks, we may want to add the polyfill recommendation to this page: https://firebase.google.com/docs/web/environments-js-sdk#polyfills

We're moving away from officially supporting IE in future versions so we won't want to include this polyfill in the SDK itself.

**Comment by sureshjoshi** [comment]
That makes sense, I also await the day when I can kick IE to the curb. The documentation update would be much appreciated though!

**Comment by DellaBitta** [comment]
I'm going to close this issue for now since we no longer support IE and there's a work around available. If you have any further questions please don't hesitate to open a new issue, and link to this one if it's related. Thanks!
